masonry_core/src/core/contexts.rs

```diff
@@ -1178,6 +1197,80 @@ impl_context_method!(
             self.global_state.mutate_callbacks.push(callback);
         }
 
+        /// Get direct reference to the stored child, and a context handle for that child.
+        ///
+        /// This context lets you set pass flags for the child widget, which may be tricky.
+        /// In general, you should avoid setting the flags of a pass that runs before the
+        /// pass you're currently in.
+        ///
+        /// See [pass documentation](crate::doc::doc_05_pass_system) for the pass order.
+        pub fn get_raw<Child: Widget + FromDynWidget + ?Sized>(
+            &mut self,
+            child: &mut WidgetPod<Child>,
+        ) -> (&Child, RawCtx<'_>) {
+            let child_state_mut = self
+                .widget_state_children
+                .item_mut(child.id())
+                .expect("get_raw: child not found");
+            let child_mut = self
+                .widget_children
+                .item_mut(child.id())
+                .expect("get_raw: child not found");
+            let child_properties = self
+                .properties_children
+                .item_mut(child.id())
+                .expect("get_raw: child not found");
+            let child_ctx = RawCtx {
+                global_state: self.global_state,
+                parent_widget_state: self.widget_state,
+                widget_state: child_state_mut.item,
+                widget_state_children: child_state_mut.children,
+                widget_children: child_mut.children,
+                properties_children: child_properties.children,
+            };
+
+            let widget = Child::from_dyn(&**child_mut.item).unwrap();
+
+            (widget, child_ctx)
+        }
+
+        /// Get direct mutable reference to the stored child, and a context handle for that child.
+        ///
+        /// This context lets you set pass flags for the child widget, which may be tricky.
+        /// In general, you should avoid setting the flags of a pass that runs before the
+        /// pass you're currently in.
```

Comment:
```suggestion
        /// pass you're currently in. No stability guarantees are made about Masonry's behaviour
        /// if this rule is not followed, and it may cause panics.
        /// 
        /// This method is an escape hatch for cases where a parent widget completely
        /// controls their child, but needs it to be a separate widget for user interaction to 
        /// behave as expected.
        /// As such, the child widget must opt-in using the `AllowRawMut` trait.
```

Reply:
In practice I think setting flags out of order isn't that bad. You can already do it with `mutate_later` and realistically, the worse it's going to do is saturate the `run_rewrite_passes()` loop every frame.

But yeah, added some details based on your text.